src/types/timezone.jl

```diff
@@ -67,32 +83,56 @@ TimeZone(::AbstractString, ::Class)
 function TimeZone(str::AbstractString, mask::Class=Class(:DEFAULT))
     # Note: If the class `mask` does not match the time zone we'll still load the
     # information into the cache to ensure the result is consistent.
-    tz, class = get!(_tz_cache(), str) do
-        tz_path = joinpath(_COMPILED_DIR[], split(str, "/")...)
+    ftz, class = get(_ftz_cache(), str, (nothing, Class(:NONE)))
+    if ftz !== nothing
+        _check_class(mask, class)
+        return ftz::FixedTimeZone
+    end
 
-        if isfile(tz_path)
-            open(TZJFile.read, tz_path, "r")(str)
-        elseif occursin(FIXED_TIME_ZONE_REGEX, str)
-            FixedTimeZone(str), Class(:FIXED)
-        elseif !isdir(_COMPILED_DIR[]) || isempty(readdir(_COMPILED_DIR[]))
-            # Note: Julia 1.0 supresses the build logs which can hide issues in time zone
-            # compliation which result in no tzdata time zones being available.
-            throw(ArgumentError(
-                "Unable to find time zone \"$str\". Try running `TimeZones.build()`."
-            ))
-        else
-            throw(ArgumentError("Unknown time zone \"$str\""))
+    vtz, class = get(_vtz_cache(), str, (nothing, Class(:NONE)))
+    if vtz !== nothing
+        _check_class(mask, class)
+        return vtz::VariableTimeZone
+    end
+
+    # We need to compute the timezone
+    tz_path = joinpath(_COMPILED_DIR[], split(str, "/")...)
+    if isfile(tz_path)
+        tz, class = open(TZJFile.read, tz_path, "r")(str)::Tuple{TimeZone,Class}
+        if tz isa FixedTimeZone
+            _ftz_cache()[str] = (tz, class)
```

Comment:
`_ftz_cache()` is now potentially called multiple times in this function, whereas previously the `_tz_cache()` was called once (since we used `get!` to insert a value if not found) ...does this affect thread-safety?